src/deepsparse/utils/onnx.py

```diff
@@ -68,6 +68,48 @@
 }
 
 
+def save_onnx_to_path(model: Model, model_path: str, external_data_file: str) -> bool:
+    """
+    Save model to the given path.  If the model has external data, store the
+    external data in 'external_data_file'.
+    Returns False if the model had no external data, True otherwise.
+    """
+    try:
+        onnx.checker.check_model(model)
+        onnx.save(model, model_path)
+        return False
+    except ValueError:
+        # Presumably the checker has failed because of external data
+        onnx.save_model(
+            model,
+            model_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=external_data_file,
+        )
+        return True
+
+
+def save_onnx(model: Model) -> str:
+    """
+    Save model to a temporary file.  Works for models with external data.
+    """
+    shaped_model = tempfile.NamedTemporaryFile(mode="w", delete=False)
+    external_data = next(tempfile._get_candidate_names())
+    has_external_data = save_onnx_to_path(model, shaped_model.name, external_data)
+
+    try:
+        yield shaped_model.name
+    finally:
+        os.unlink(shaped_model.name)
+        shaped_model.close()
+        if has_external_data:
+            external_data_path = os.path.join(
+                os.path.dirname(shaped_model.name), external_data
+            )
+            os.unlink(external_data_path)
```

Comment:
i'm not sure how this will interact with the `@contextlib.contextmanager` decorator in the consumer function.. maybe this function needs the decorator as well and the consumers should also `yield` the return value?

Reply:
for the record if i add @contextlib.contextmanager to save_onnx then i get an error.  if i remove it from the functions that are now calling it then the error goes away.   so i'm guessing the best thing is to just move this annotation to save_onnx?

Reply:
Also unsure, it's definitely hard to reason about lol. It might be easier to refactor all these functions to act directly on onnx models and handle the saving logic directly in the pipelines where these are called. But that's a wider change than the scope of this PR at least.

> if i remove it from the functions that are now calling it then the error goes away.

Was this with contextmanager + yielding the return value was crashing?

Reply:
yeah, if i tagged all the functions with that, it would error out.  things work if only save_onnx is tagged or if the immediate "override..." callers are tagged.